### ISSUE TITLE
bug(aniamtions) nested animations don't work properly when switching between states

### ISSUE THREAD

**Issue by Ploppy3** [angular/angular#25194]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior
Nested animations don't work properly when switching between states

## Expected behavior
Nested animations should always work.

## Minimal reproduction of the problem with instructions
Working (without swithing between states): https://stackblitz.com/edit/angular-sev2pq?file=src%2Fapp%2Fapp.component.html

Not working (switching between states): https://stackblitz.com/edit/angular-skyqmo?file=src%2Fapp%2Fapp.component.html

## Environment

<pre><code>
Angular version: X.Y.Z

Browser:
- [x] Chrome (desktop) version 68
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX

Others:

</code></pre>

**Comment by dario-piotrowicz** [comment]
This seems to be an issue with the height `*` not being correctly recognised, I've checked and I think this bug is going to be fixed by https://github.com/angular/angular/pull/44515 (I've checked locally and the PR does seem to fix it)

I'll leave another comment here after/if the PR gets merged and after having checked if the PR's changes do indeed fix this :slightly_smiling_face::+1:

**Comment by dario-piotrowicz** [comment]
@Ploppy3 yeah the issue's been fixed as you can check here: https://stackblitz.com/edit/angular-ivy-u9vwei?file=src%2Fapp%2Fapp.component.html :smile:

**Comment by Ploppy3** [comment]
@dario-piotrowicz Congrats !

**Comment by thePunderWoman** [comment]
This issue was closed as obsolete due to the introduction of the new `animate.enter` and `animate.leave` APIs. The `@angular/animations` package is now deprecated and will be removed in Angular v23. We hope you give the new APIs a try.
